extractor/src/main/java/com/example/extractor/ConfigSerializer.java

```diff
@@ -99,13 +101,21 @@ public static String serializeAllInOneConfigToString(final String[] exampleFileP
 
     private static Configuration addIdProperty(final Configuration config, final String idValue) {
         return new Configuration() {
+            /** Use a LinkedHashSet to maintain the order of property names and add "id" */
+            private final Set<String> propertyNames = new LinkedHashSet<>(Arrays.asList(config.getPropertyNames()));
+
+            {
+                propertyNames.add("id");
+            }
```

Comment:
Still a hack.

Please use this class and this method to have configuration and add property to it
https://github.com/checkstyle/checkstyle/blob/d2c151213399be206114ed6ed318dcf4bc5c7922/src/main/java/com/puppycrawl/tools/checkstyle/DefaultConfiguration.java#L152